app/actions/apps.ts

```diff
@@ -3,65 +3,79 @@
 
 import {Client4} from '@mm-redux/client';
 import {ActionFunc} from '@mm-redux/types/actions';
-import {AppCallResponse, AppForm, AppCallRequest} from '@mm-redux/types/apps';
+import {AppCallResponse, AppForm, AppCallRequest, AppCallType} from '@mm-redux/types/apps';
 import {AppCallTypes, AppCallResponseTypes} from '@mm-redux/constants/apps';
-import {sendEphemeralPost} from './views/post';
 import {handleGotoLocation} from '@mm-redux/actions/integrations';
 import {showModal} from './navigation';
 import {Theme} from '@mm-redux/types/preferences';
 import CompassIcon from '@components/compass_icon';
 import {getTheme} from '@mm-redux/selectors/entities/preferences';
 import EphemeralStore from '@store/ephemeral_store';
+import {makeCallErrorResponse} from '@utils/apps';
 
-export function doAppCall<Res=unknown>(call: AppCallRequest, intl: any): ActionFunc {
+export function doAppCall<Res=unknown>(call: AppCallRequest, type: AppCallType, intl: any): ActionFunc {
     return async (dispatch, getState) => {
-        const ephemeral = (text: string) => dispatch(sendEphemeralPost(text, call?.context.channel_id, call?.context.root_id));
         try {
-            const res = await Client4.executeAppCall(call) as AppCallResponse<Res>;
+            const res = await Client4.executeAppCall(call, type) as AppCallResponse<Res>;
             const responseType = res.type || AppCallResponseTypes.OK;
 
             switch (responseType) {
             case AppCallResponseTypes.OK:
-                if (res.markdown) {
-                    if (!call.context.channel_id) {
-                        return {data: res};
-                    }
-
-                    ephemeral(res.markdown);
-                }
                 return {data: res};
             case AppCallResponseTypes.ERROR:
-                return {data: res, error: {message: res.error}};
+                return {data: res};
```

Comment:
This part of the code is dealing all the time with the AppCallResponse structure. The AppCallResponse already has an error type and field and may have extra context for the element that will handle the error. I understand this is not very semantic, but dividing here (and potentially in many other places down the road) the structure to handle Errors and the rest of the results separately will generate more confusion, I think.

Reply:
> the structure to handle Errors and the rest of the results separately will generate more confusion, I think.

I agree, though per PR feedback I did start using the `error` return value in some places where `doAppCall` is called, to try to conform to the rest of the mobile app's error handling. So those instances (particularly in the apps modal form) will need to be reverted to avoid using that field and instead use the call response's error field.

Reply:
you decide what is best but obviously this Pr is impacting the changes done in the other PR. In my opinion it should follow the `{data: ..., error: ... }` pattern and I don't see why it should change unless there is a better reason than having to deconstruct the response from the server.